Initial containers + NavBar + watertracker prototype

taylorsmart opened the pull request:
- Created general containers on the index.js for key dashboard page components
- Added nav bar up top (not formatted aesthetically)
- Added water tracker prototype functionality

NOTE: The watertracker prototype is intentionally a switch statement in order to allow more flexibility with adding a variety of different container assets down the road.

Commit:
Committing changes prior to merge

Commit:
merge commit

Scott-Guinn commented on components/caloricHistory.jsx:
```diff
@@ -0,0 +1,15 @@
+  );
+};
+
+export default WaterHistory;
```
Is there a reason this file is named caloricHistory.jsx but the component is named WaterHistory?

emmaknor replied:
good question

taylorsmart replied:
Error - was creating a storage location and forgot to change it.  Appreciate it!

Scott-Guinn commented on components/dailyTracker.jsx:
```diff
@@ -0,0 +1,14 @@
+  );
+};
+
+export default DailyTracker;
```
can we match casing in the file names as well as per react convention? "DailyTracker.jsx"

Scott-Guinn commented on components/waterHistory.jsx:
```diff
@@ -0,0 +1,57 @@
+import styles from '../styles/Home.module.css';
+
+const WaterHistory = () => {
+  const generateWaterGraph = (waterDrank = 62, waterGoal = 100) => {
```
Are these default values in here just for testing? If so, can we put a comment that says that so people know.

emmaknor replied:
Additionally, we shouldn't need default values if the input validation from the data entry team is set up correctly, but if this is for testing/initialization purposes no worries!

Scott-Guinn commented on components/waterHistory.jsx:
```diff
@@ -0,0 +1,57 @@
+  const generateWaterGraph = (waterDrank = 62, waterGoal = 100) => {
+    const waterPercentage = waterGoal > 0 ?
+      Math.round((waterDrank / waterGoal) * 10) :
+      0;
```
when would the waterGoal be less than/equal to 0?

Scott-Guinn replied:
I would argue that this is redundant if we validate on entry and makes the code less readable.

taylorsmart committed:
Update .eslintrc.js

Scott-Guinn commented on components/waterHistory.jsx:
```diff
@@ -0,0 +1,57 @@
+      0;
+
+    let remainder = 0;
+    remainder = waterPercentage - remainder;
```
Might need to hop in a DC channel with you and have you step me through this. Can we combine these two lines and just set remainder = waterPercentage? Why are we subtracting 0 from it?

emmaknor replied:
I too don't fully understand when the remainder is updated. Will this be stored as state?

Scott-Guinn commented on components/waterHistory.jsx:
```diff
@@ -0,0 +1,57 @@
+    remainder = waterPercentage - remainder;
+    const bottleArray = [];
+    for (let j = 0; j < 10; j += 1) {
+      switch (remainder) {
```
for future implementation, I think best practices would be to use an if statement now and refactor to a switch down the road when it makes sense to do so. But lets see what your second reviewer thinks.

emmaknor replied:
As of now it makes more sense to me to use an if statement; however, I do not know what your plans are for adding more containers as you stated previously. I am fine with this staying a switch statement for the time being and re-evaluating when the additional containers are implemented.

emmaknor commented on pages/index.js:
```diff
@@ -24,8 +28,40 @@ const App = () => {
         <link rel="icon" href="/favicon.ico" />
       </Head>
-
+      <Navbar 
```
Just to confirm as far as styling goes, we all plan to implement a dark theme for our MVP and then the light theme is our stretch goal?

Scott-Guinn replied:
I found dark/light theme to be exceptionally easy to implement in FEC so I don't think it's too much of a stretch to say we'd do both for MVP. Typically light theme is the standard so maybe we want to do that first.

emmaknor commented on pages/index.js:
```diff
@@ -24,8 +28,40 @@ const App = () => {
+        style={{display: 'flex', justifyContent: 'space-between'}}>
+        <Navbar.Brand href="#home">
+          <strong
+            style={{
```
to keep styling consistent & stored in the same place it may be best to avoid inline styles & keep as much of our css as possible in the styles folder. it also may be a good idea to agree on some basic styling as a group so that our app stays cohesive as we flesh out our components.

Scott-Guinn replied:
^ I'm a culprit of this too, although for testing and development sometimes it's easier to use in-line styling and then go back and move it to a css file later.

taylorsmart committed:
Updating class name

taylorsmart committed:
Testing over

Setting default to 0 and 100

taylorsmart committed:
Removed styling for consistency

taylorsmart committed:
Updated the ridiculous subtract from zero I had... 

Thanks for catching that @Scott-Guinn and @emmaknor .

Outlined the logic behind the water percentage detriment + examples on how I'll implement the switch statement.
